NetKet/Machine/ffnn.hpp

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Eigen/Dense>
+#include <iostream>
+#include <vector>
+#include "Lookup/lookup.hpp"
+#include "Utils/all_utils.hpp"
+#include "layer.hpp"
+
+#ifndef NETKET_FFNN_HPP
+#define NETKET_FFNN_HPP
+
+namespace netket {
+
+template <typename T>
+class FFNN : public AbstractMachine<T> {
+  using VectorType = typename AbstractMachine<T>::VectorType;
+  using MatrixType = typename AbstractMachine<T>::MatrixType;
+
+  std::vector<std::unique_ptr<AbstractLayer<T>>>
```

Comment:
In `layer.hpp` you add a typedef `using Ptype = unique_ptr<AbstractLayer<T>>`. Maybe do it here as well.

Reply:
Ok